### PR TITLE
cmake refactor/document init

### PR DESCRIPTION
**Pull request by Dekken**
suggestion in tandem to https://github.com/PHAREHUB/PHARE/issues/270

### CODE REVIEW

File: res/cmake/def.cmake [hunk 1]
```diff
@@ -1,116 +1,42 @@
...
 
 set (PHARE_HAS_HIGHFIVE "0")
 if (HighFive)
  set (PHARE_HAS_HIGHFIVE "1")
 endif()
 
-# Pybind errors with clang, it is default in GCC
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-  set (CMAKE_CXX_FLAGS -fsized-deallocation)
+if (NOT DEFINED PHARE_MPI_PROCS)
```
Comment (nicolasaunai) [inline res/cmake/def.cmake:12]: shouldn' that rather be in the test config file?

File: res/cmake/def.cmake [hunk 2]
```diff
@@ -1,116 +1,42 @@
...
+  set(PHARE_MPI_PROCS 1)
+  if(testMPI)
+    set(PHARE_MPI_PROCS 2)
+  endif()
 endif()
 
-if(coverage AND NOT MSVC)
-  set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pg -DHAVE_EXECINFO_H -g3")
+# Pybind errors with clang, it is default in GCC
```
Comment (nicolasaunai) [inline res/cmake/def.cmake:19]: if that's for pybind shouldn't that rather be in the pybind config file?

File: res/cmake/func.cmake [hunk 3]
```diff
@@ -0,0 +1,72 @@
+
```
Comment (nicolasaunai) [inline res/cmake/func.cmake:1]: does it make sense actually to have a "function" file where all functions are ? wouldn't it be easier to find these functions with the config they are associated with ? like those would go in test cmake ?

File: res/cmake/coverage.cmake [hunk 4]
```diff
@@ -1,38 +1,38 @@
...
-      endif(APPLE)
-
-      add_custom_target(show_coverage
-          COMMAND ${OPPEN_CMD} ${CMAKE_CURRENT_BINARY_DIR}/coverage/index.html
-          DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/coverage/index.html gcovr
-          )
-  ENDIF(coverage)
-
-ENDIF(test)
+if (test AND coverage)
```
Comment (nicolasaunai) [inline res/cmake/coverage.cmake:2]: we need more comments for the "whys" we do each of these nasty configs....

File: src/core/numerics/ohm/ohm.h [hunk 5]
```diff
@@ -183,8 +183,8 @@ namespace core
 
 
         template<typename Field, typename ComponentTag>
-        auto pressure_(Field const& n, Field const& Pe, MeshIndex<Field::dimension> index,
-                       ComponentTag) const
+        auto pressure_(Field const& n, Field const& Pe,
+                       [[maybe_unused]] MeshIndex<Field::dimension> index, ComponentTag) const
```
[the remark this answers is not in the document]
Comment (Dekken) [reply src/core/numerics/ohm/ohm.h:187]: pretty sure it's werror
not sure in what case this function is called and it's never used, that could be a bug

File: res/cmake/def.cmake (hunk 1 above)
Comment (Dekken) [reply res/cmake/def.cmake:12]: it's something used by testing, but it's it only used by testing?
currently maybe, so maybe yes it should be

File: res/cmake/func.cmake (hunk 3 above)
Comment (Dekken) [reply res/cmake/func.cmake:1]: if they all have a file they belong to it makes sense
is there any orphans?

File: res/cmake/coverage.cmake (hunk 4 above)
Comment (Dekken) [reply res/cmake/coverage.cmake:2]: yes, I will probably do what I did for werror and split it up

File: src/core/numerics/ohm/ohm.h (hunk 5 above)
[the remark this answers is not in the document]
Comment (Dekken) [reply src/core/numerics/ohm/ohm.h:187]: ignore it, probably shouldn't be in this PR

File: res/cmake/def.cmake (hunk 2 above)
Comment (Dekken) [reply res/cmake/def.cmake:19]: I'm not sure it's smart to separate setting variables like CMAKE_CXX_FLAGS all that much

Comment (Dekken) [reply res/cmake/def.cmake:19]: perhaps it should be "PHARE_FLAGS" instead of CMAKE_CXX_FLAGS

**Comment by Dekken** [comment]
not sure if the symlink handling should be done in CMake but I think it might better than accidental infinite loops.

it does mean that for #309 it would need a "git reset -hard" before "git pull" I think

File: res/cmake/coverage.cmake (hunk 4 above)
Comment (Dekken) [reply res/cmake/coverage.cmake:2]: I removed a lot of flags that were (I believe) unnecessary for the point of enabling coverage
so it should be a bit simpler now

**Pull request merged**